File: flutter_app/lib/models/ai_chat.dart
/// AI 间私聊记录（Phase 1 只读展示）
class AIChat {
  final int id;
  final int characterAId;
  final String characterAName;
  final int characterBId;
  final String characterBName;
  final int speakerId;
  final String speakerName;
  final int roundSeq;
  final String content;
  final DateTime createdAt;

  const AIChat({
    required this.id,
    required this.characterAId,
    required this.characterAName,
    required this.characterBId,
    required this.characterBName,
    required this.speakerId,
    required this.speakerName,
    required this.roundSeq,
    required this.content,
    required this.createdAt,
  });

  factory AIChat.fromJson(Map<String, dynamic> json) {
    return AIChat(
      id: (json['id'] as num).toInt(),
      characterAId: (json['character_a_id'] as num).toInt(),
      characterAName: json['character_a_name'] as String? ?? '',
      characterBId: (json['character_b_id'] as num).toInt(),
      characterBName: json['character_b_name'] as String? ?? '',
      speakerId: (json['speaker_id'] as num).toInt(),
      speakerName: json['speaker_name'] as String? ?? '',
      roundSeq: (json['round_seq'] as num).toInt(),
      content: json['content'] as String? ?? '',
      createdAt:
          DateTime.tryParse(json['created_at'] as String? ?? '')?.toLocal() ??
              DateTime.now(),
    );
  }
}

File: backend/app/models/chat/ai_chat.py
"""AI 间私聊记录（Phase 1）：同用户两个 AI 角色私下对话，落库供只读展示"""
from datetime import datetime

from sqlalchemy import Integer, DateTime, ForeignKey, func, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class AIChat(Base):
    __tablename__ = "ai_chats"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    character_a_id: Mapped[int] = mapped_column(Integer, ForeignKey("ai_characters.id"), nullable=False, index=True)
    character_b_id: Mapped[int] = mapped_column(Integer, ForeignKey("ai_characters.id"), nullable=False, index=True)
    speaker_id: Mapped[int] = mapped_column(Integer, nullable=False)  # 发言角色 id（a 或 b）
    round_seq: Mapped[int] = mapped_column(Integer, default=0)  # 同事件内轮次（0 起，0=事件首条）
    content: Mapped[str] = mapped_column(Text, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now(), index=True)
